src/ServiceControl.Audit/Auditing/AuditPersister.cs

```diff
@@ -182,6 +182,7 @@ static void RecordKnownEndpoints(EndpointDetails observedEndpoint, Dictionary<st
                     HostId = observedEndpoint.HostId,
                     LastSeen = processedMessage.ProcessedAt,
                     Name = observedEndpoint.Name,
+                    NServiceBusVersion = observedEndpoint.NServiceBusVersion,
```

Comment:
What's the context of this code? Is `observedEndpoints` a collection of only the endpoints observed during the batch being ingested, and then updates to known endpoints are done at the same time as the batch is ingested? Because otherwise I don't understand how `LastSeen` would be updated, let alone `NServiceBusVersion`.

What happens if you are doing a phased update of an endpoint's instances, and one endpoint instance is updated to a new NServiceBus version and another is still running the old. Will the observed value in KnownEndpoints oscillate?

Reply:
Yea that code is funky, need to understand this better if we ever go ahead with this